apikey-service/src/main/java/eu/europeana/apikey/controller/ApikeyController.java

```diff
@@ -142,6 +144,12 @@ public ApikeyController(ApikeyRepo apikeyRepo, CaptchaManager captchaManager, Cu
             LOG.debug(e.getMessage() + ", abort creating apikey");
             return new ResponseEntity<>(e.getMessage(), HttpStatus.BAD_REQUEST);
         }
+        try{
+            apiKeyAlreadyExist(apikeyCreate);
+        } catch (ApikeyException e) {
+            LOG.debug("Api key already exist for " + e.getMessage());
+            return new ResponseEntity<>(e.getMessage(), HttpStatus.BAD_REQUEST);
+        }
```

Comment:
Exceptions should only be thrown if there is a problem from which you can't recover. They should not be used in a 'normal' flow of events. In this case I feel it would be better to let the apiKeyAlreadyExists return a boolean and check that. That is more efficient than throwing and catching an error.